plugins/modules/discovery_intent.py

```diff
@@ -1338,17 +1343,62 @@ def get_task_status(self, task_id=None):
                 self.log(msg, "CRITICAL")
                 self.module.fail_json(msg=msg)
                 return False
+
             self.log("Task status for the task id (before checking status) {0} is {1}".format(str(task_id), str(response)), "INFO")
             progress = response.get('progress')
-            if progress in ('In Progress', 'Inventory service initiating discovery'):
+            try:
+                progress_value = int(progress)
+                result = True
+                self.log("The Process is completed", "INFO")
+                self.result.update(dict(discovery_task=response))
+                return result
+            except Exception:
+                self.log("The progress status is {0}, continue to check the status after 3 seconds. Putting into sleep for 3 seconds".format(progress))
                 time.sleep(3)
-                continue
-            else:
+
+    def get_deleted_task_status(self, task_id=None):
+        """
+        Monitor the status of a task of deletion of dicovery in the Cisco Catalyst Center.
+        It checks the itask status periodically until the task is 'Discovery deleted successfully'.
+        If the task encounters an error or fails, it immediately fails the module and returns False.
+
+        Parameters:
+          - task_id: The ID of the task to monitor.
+
+        Returns:
+          - result: True if the task completed successfully, False otherwise.
+        """
+
+        result = False
+        params = dict(task_id=task_id)
+        while True:
+            response = self.dnac_apply['exec'](
+                family="task",
+                function='get_task_by_id',
+                params=params,
+                op_modifies=True,
+            )
+            response = response.response
+            self.log("Task status for the task id {0} is {1}, is_error: {2}".format(str(task_id), str(response), str(response.get('isError'))), "INFO")
+            if response.get('isError') or re.search(
+                'failed', response.get('progress'), flags=re.IGNORECASE
+            ):
+                msg = 'Discovery task with id {0} has not completed - Reason: {1}'.format(
+                    task_id, response.get("failureReason"))
+                self.log(msg, "CRITICAL")
+                self.module.fail_json(msg=msg)
+                return False
+
+            self.log("Task status for the task id (before checking status) {0} is {1}".format(str(task_id), str(response)), "INFO")
+            progress = response.get('progress')
+            if re.search('Discovery deleted successfully.', response.get('progress')):
                 result = True
                 self.log("The Process is completed", "INFO")
```

Comment:
Can we add more information here? 
Discovery deleted successfully ... something more meaningful?